### PR TITLE
RavenDB-6840 Implement encryption key management

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 1]
```diff
@@ -104,6 +104,8 @@ public Task Get()
         {
             var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
             var node = GetStringQueryString("node", false);
+            var encrypted = GetBoolValueQueryString("encrypted", required: false) ?? false;
```
Comment (ayende) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs:107]: Remove this, we are ALWAYS encrypted if the db is encrypted, or we are never encrypted if the db is not encrypted.
In other words, you can only specify encryption during db creation, not when adding a node

File: src/Raven.Client/Server/DatabaseRecord.cs [hunk 2]
```diff
@@ -25,6 +25,8 @@ public DatabaseRecord(string databaseName)
 
         public bool Disabled;
 
+        public bool Encrypted;
```
Comment (ayende) [inline src/Raven.Client/Server/DatabaseRecord.cs:28]: Where are you actually enforcing this?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 3]
```diff
@@ -289,8 +293,162 @@ public void Initialize()
...
+            Debug.Assert(context.Transaction != null);
+
+            var record = Cluster.ReadDatabase(context, name);
+
+            if (record != null)
+                throw new InvalidOperationException($"Cannot delete key {name} where there is an existing database that require its usage");
+
+            var tree = context.Transaction.InnerTransaction.CreateTree("SecretKeys");
 
-            Task.Run(ClusterMaintanceSetupTask, ServerShutdown);
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ServerStore.cs:293]: why did you remove this line?

File: src/Raven.Client/Server/DatabaseRecord.cs (hunk 2 above)
Comment (iftahbe) [reply src/Raven.Client/Server/DatabaseRecord.cs:28]: Will fix, http://issues.hibernatingrhinos.com/issue/RavenDB-6916

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs (hunk 1 above)
Comment (iftahbe) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs:107]: Yes, will remove.

File: src/Raven.Server/ServerWide/ServerStore.cs (hunk 3 above)
Comment (iftahbe) [reply src/Raven.Server/ServerWide/ServerStore.cs:293]: Weird mistake, will fix
